b9lab-content/3-main-concepts/15-migrations.md

```diff
@@ -1,5 +1,81 @@
 # Migrations
 
+## On-chain Upgrades
+
+A Cosmos SDK application running on a Cosmos blockchain can be upgraded in an orderly, on-chain fashion.
+
+Upgrading blockchains and blockchain applications is notoriously difficult and risky. Cosmos SDK solves for the common risks and challenges. What are those challenges and risks that Cosmos SDK solves?
+
+Generally, when a blockchain is upgraded it is vital that all nodes upgrade simultaneously and at the same block height. In a disorderly setting, this is difficult to actually achieve. If the nodes do not do so then the blockchain will "fork" into two blockchains with common history - one chain that observes the new rules and one chain that observes the old rules. It is generally not possible for the two chains to reach a common consensus or merge together in the future.
+
+<HighlightBox type="info">
+
+Without software support for upgrades, upgrading a live chain is risky because all of the validators need to pause their state machines at exactly the same block height and apply the upgrade before resuming. If this is not done correctly, there can be state inconsistencies which are hard to recover from.
+
+</HighlightBox>
+
+Smart contracts on EVM chains such as Ethereum are immutable software. By definition, they are difficult or impossible to change. Various strategies based on modularity can simulate the effects of upgrading the contracts but all known methods have inherent limitations. Chief among the limitations are the difficulties, impossibility or prohibitive cost of reorganizing data at rest. This places a significant limitation on the types of upgrades that are feasible.
+
+A Cosmos SDK blockchain built for a specific application can be upgraded without forks and, if necessary, the existing data can be reorganized to prepare it for use by a new version of the application and blockchain.
```

Comment:
could you please rephrase  " reorganized to prepare it for use by a new version of the application and blockchain" to make it more clear.😁

Reply:
A Cosmos SDK blockchain built for a specific application can be upgraded without forks. In the case that a new version of the blockchain application uses a different data layout t han exists on chain, the existing data can be reorganized before the new version of the application comes online. The data migration is defined by the developer and runs in each node, quickly and cost-effectively before the node returns to service.